src/vivarium_public_health/risks/observer.py

```diff
@@ -0,0 +1,80 @@
+import pandas as pd
+from vivarium_public_health.risks.data_transformation import split_risk_from_type
+
+
+class CategoricalRiskObserver:
+    """ An observer for a categorical risk factor.
+    This component by default observes proportion of simulants in each age
+    group who are alive and in each category of risk at the midpoint of a year
+    unless the sample date is specified in the configuration.
+
+    It also collects the total number of alive simulants in each age group
+    when the proportion is collected.
+
+    """
+    configuration_defaults = {
+        'observer': {
+            'risk': {
+                'by_year': True,
+                'sample_date': {
+                    'month': 7,
+                    'day': 1
+                }
+            }
+        }
+    }
+
+    def __init__(self, risk: str):
+        """
+        Parameters
+        ----------
+        risk :
+        the type and name of a risk, specified as "type.name". Type is singular.
+
+        """
+        self.risk_type, self.risk_name = split_risk_from_type(risk)
+        self.configuration_defaults = CategoricalRiskObserver.configuration_defaults
+
+    def setup(self, builder):
+        self.clock = builder.time.clock()
+        self.population_view = builder.population.get_view(['alive', 'age'])
+        self.age_bins = builder.data.load('population.age_bins')
```

Comment:
good idea to use age bins to get the age group name - have we updated the mortality observer to do the same?